electrum/gui/qt/__init__.py

```diff
@@ -398,26 +401,65 @@ def start_new_window(
         return window
 
     def _start_wizard_to_select_or_create_wallet(self, path) -> Optional[Abstract_Wallet]:
-        wizard = InstallWizard(self.config, self.app, self.plugins, gui_object=self)
-        try:
-            path, storage = wizard.select_storage(path, self.daemon.get_wallet)
-            # storage is None if file does not exist
-            if storage is None:
-                wizard.path = path  # needed by trustedcoin plugin
-                wizard.run('new')
-                storage, db = wizard.create_storage(path)
-            else:
-                db = WalletDB(storage.read(), storage=storage, manual_upgrades=False)
-                wizard.run_upgrades(storage, db)
-        except (UserCancelled, GoBack):
-            return
-        except WalletAlreadyOpenInMemory as e:
-            return e.wallet
-        finally:
-            wizard.terminate()
-        # return if wallet creation is not complete
-        if storage is None or db.get_action():
+        wizard = QENewWalletWizard(self.config, self.app, self.plugins, self.daemon, path)
+        result = wizard.exec()
+        # TODO: use dialog.open() instead to avoid new event loop spawn?
+        self.logger.info(f'{result}')
+        if result == QENewWalletWizard.Rejected:
+            self.logger.info('ok bye bye')
             return
+
+        d = wizard.get_wizard_data()
+
+        if d['wallet_is_open']:
+            for window in self.windows:
+                if window.wallet.storage.path == d['wallet_name']:
+                    return window.wallet
+            raise Exception('found by wizard but not here?!')
+
+        if not d['wallet_exists']:
+            self.logger.info('about to create wallet')
+            wizard.create_storage()
+            if d['wallet_type'] == '2fa' and 'x3/' not in d:
+                return
+            wallet_file = wizard.path
+        else:
+            wallet_file = d['wallet_name']
+
+        storage = WalletStorage(wallet_file)
+        if storage.is_encrypted_with_user_pw() or storage.is_encrypted_with_hw_device():
+            storage.decrypt(d['password'])
```

Comment:
`InvalidPassword` should be handled somewhere
```
 28.11 | E | gui.qt.ElectrumGui | 
Traceback (most recent call last):
  File "...\electrum\electrum\gui\qt\__init__.py", line 362, in start_new_window
    wallet = self._start_wizard_to_select_or_create_wallet(path)
  File "...\electrum\electrum\gui\qt\__init__.py", line 431, in _start_wizard_to_select_or_create_wallet
    storage.decrypt(d['password'])
  File "...\electrum\electrum\storage.py", line 172, in decrypt
    s = zlib.decompress(ec_key.decrypt_message(self.raw, enc_magic))
  File "...\electrum\electrum\ecc.py", line 546, in decrypt_message
    raise InvalidPassword()
electrum.util.InvalidPassword: Incorrect password
```

Reply:
Note: applies to both mistyped user pw, and mismatching hardware device cases

Reply:
added in 7940c0725ebfb17b527cc7b95311a76364444740